### PR TITLE
modify setDateRange to set start date first

### PR DESCRIPTION
**Pull request by ananzh**
Issue Resolved:
https://github.com/opensearch-project/opensearch-dashboards-test-library/issues/14

 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ananzh** [comment]
By changing params order, this change will not break any current customer usage. For OSD core, 2.0, bwc with security, test results:

* check_advanced_settings
Screen Shot 2022-04-29 at 22 19 24

* check_default_page
Screen Shot 2022-04-29 at 22 20 02

* check_filter_and_query
Screen Shot 2022-04-29 at 22 20 45

* check_loaded_data
Screen Shot 2022-04-29 at 22 22 19

* check_timeline
Screen Shot 2022-04-29 at 22 25 01

**Review by kavilla — APPROVED** [review]

**Comment by kavilla** [comment]
Do we need to bump this? https://github.com/opensearch-project/opensearch-dashboards-test-library/blob/main/package.json#L3

File: common-utils/common-UI/common-UI.js [hunk 1]
```diff
@@ -16,22 +16,23 @@ export class CommonUI {
 
   /**
    * Sets the page date range
+   * Sets start date first then end date
    * @param {String} start Start datetime to set
    * @param {String} end  End datetime to set
    */
-  setDateRange(start, end) {
+  setDateRange(end, start) {
```
Comment (seraphjiang) [inline common-utils/common-UI/common-UI.js:23]: any reason we break the api contract?

**Comment by seraphjiang** [comment]
> Do we need to bump this? https://github.com/opensearch-project/opensearch-dashboards-test-library/blob/main/package.json#L3

@tianleh it looks we need a new version, could you confirm

File: common-utils/common-UI/common-UI.js (hunk 1 above)
Comment (tianleh) [reply common-utils/common-UI/common-UI.js:23]: I think it is a tradeoff between keeping the general practice (as @seraphjiang mentioned above) and minimizing the amount of changes across multiple repos.

The current [consumers](https://github.com/search?q=org%3Aopensearch-project+setDateRange&type=code) of this repo are using the API in the end date first way. Thus this PR won't need them to make corresponding changes.

**Comment by tianleh** [comment]
> > Do we need to bump this? https://github.com/opensearch-project/opensearch-dashboards-test-library/blob/main/package.json#L3
> 
> @tianleh it looks we need a new version, could you confirm

Yes. Please bump the patch version. @ananzh @kavilla @seraphjiang 

Thanks for this fix!

**Comment by tianleh** [comment]
I created an issue to enhance the PR check workflows for this repo https://github.com/opensearch-project/opensearch-dashboards-test-library/issues/17

**Comment by kavilla** [comment]
I added the version bump and backport workflow here: https://github.com/opensearch-project/opensearch-dashboards-test-library/pull/18. If this PR gets merged prior to the label might need to be removed and re-added to this PR to trigger the workflow.

File: common-utils/common-UI/common-UI.js (hunk 1 above)
Comment (Tengda-He) [reply common-utils/common-UI/common-UI.js:23]: not sure why need to reverse the order in the requirement, 

just my two cents if the assumption is that end  always greater start,  then we can implement the validation inside the setDateRange method:

if (end < start) {
    start, end = end, start
}

**Review by Tengda-He — APPROVED** [review]

**Review by tianleh — APPROVED** [review]

**Pull request merged**